xdsl/dialects/mesh.py

```diff
@@ -1,17 +1,113 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+from enum import auto
+from typing import TypeAlias
+
 from xdsl.dialects.builtin import (
+    I16,
     I64,
+    ArrayAttr,
+    BytesAttr,
     DenseArrayBase,
+    FlatSymbolRefAttr,
+    IntegerAttr,
     SymbolNameConstraint,
+    i16,
+    i64,
+)
+from xdsl.dialects.utils import DimensionList, DynamicIndexList
+from xdsl.ir import (
+    Attribute,
+    Dialect,
+    EnumAttribute,
+    OpaqueSyntaxAttribute,
+    ParametrizedAttribute,
+    SpacedOpaqueSyntaxAttribute,
+    TypeAttribute,
+    VerifyException,
 )
-from xdsl.dialects.utils.dimension_list import DimensionList
-from xdsl.ir import Dialect, VerifyException
 from xdsl.irdl import (
+    AttrSizedOperandSegments,
     IRDLOperation,
+    irdl_attr_definition,
     irdl_op_definition,
+    opt_prop_def,
     prop_def,
+    result_def,
     traits_def,
+    var_operand_def,
 )
-from xdsl.traits import SymbolOpInterface
+from xdsl.parser import AttrParser
+from xdsl.printer import Printer
+from xdsl.traits import Pure, SymbolOpInterface
+from xdsl.utils.hints import isa
+from xdsl.utils.str_enum import StrEnum
+
+MeshAxesAttr: TypeAlias = DenseArrayBase[I16]
+
+
+@irdl_attr_definition
+class MeshAxesArrayAttr(ParametrizedAttribute, OpaqueSyntaxAttribute):
+    name = "mesh.axisarray"
+
+    axes: ArrayAttr[MeshAxesAttr]
+
+    @classmethod
+    def parse_parameters(cls, parser: AttrParser) -> Sequence[Attribute]:
+        axes = parser.parse_comma_separated_list(
+            parser.Delimiter.SQUARE,
+            parser.parse_attribute,
```

Comment:
feels like this could be parse_integer?

But could also abstract this out into a function that parses a DenseArrayBase[I16] and parse a list of those directly?
```python
                elements = parser.parse_comma_separated_list(
                    parser.Delimiter.SQUARE, parser.parse_integer
                )
            return DenseArrayBase(
                self.elt_type, BytesAttr(self.elt_type.pack(elements))
            )
```